### DISCUSSION TITLE
DS18B20 False value of -2029.9℃  - How to prevent

### DISCUSSION THREAD

**Discussion by garyvdm** [arendst/Tasmota#21737] [General]
I have a [DS18B20](https://tasmota.github.io/docs/DS18x20/) temperature sensor, which I use in some rules to make a simple thermostat.  

I have mqtt and home assistant setup, so I can monitor.

Occasionally, the sensor reports  -2029.9℃, which anyone that has done high school physics will know is impossible.

What might be causing this false value, and want can be done to prevent it from happening?

![Screenshot from 2024-07-05 17-58-27](https://github.com/arendst/Tasmota/assets/352175/47d9783a-3c3d-45b7-b2b3-ff69a1acfd72)

**Comment by sfromis** [comment]
IDK, maybe there is electrical noise screwing up signals, including power issues. Or you are using some bad DS18B20 clone (nearly all sold are dubious China-clones).

**Reply by garyvdm** [reply]
Ok, thanks.

Is there a way to filter out values that are not within a reasonable range?

**Reply by sfromis** [reply]
Of course, you can apply a filter where the data is being used. If you instead mean to not have Tasmota report what you get from the sensor, no such filtering available.

`SetOption126` exists if you want to experiment, but I'm not convinced.....
> Enable arithmetic mean over teleperiod for JSON temperature for DS18x20 sensors

**Comment by lalo-uy** [comment]
Since you are using rules, you can catch the odd value there and discart it.

**Reply by sfromis** [reply]
Except that it will not be discarded from MQTT or display.

**Comment by lalo-uy** [comment]
You could publish the good ones on a different topic. Ok with the wrong
display.
